src/toil/job.py

```diff
@@ -3263,6 +3268,39 @@ def _resolve(cls, jobStoreLocator, jobStoreFileID):
             value = safeUnpickleFromStream(fileHandle)
             return value
 
+# Machinery for type-safe-ish Toil Python workflows.
+#
+# TODO: Until we make Promise generic on the promised type, and work out how to
+# tell MyPy that rv() returns a Promise for whatever type that object's run()
+# method returns, this won't actually be type-safe, because any Promise will be
+# a Promised[] for any type.
+
+T = TypeVar('T')
+# We have type shorthand for a promised value.
+# Uses a generic type alias, so you can have a Promised[T]. See <https://github.com/python/mypy/pull/2378>.
+
+Promised = Union[Promise, T]
+
+def unwrap(p: Promised[T]) -> T:
+    """
+    Function for ensuring you actually have a promised value, and not just a promise.
+    Mostly useful for satisfying type-checking.
+
+    The "unwrap" terminology is borrowed from Rust.
+    """
+    assert not isinstance(p, Promise)
```

Comment:
```suggestion
    if not isinstance(p, Promise):
        raise AssertionError(f"{p!r} is not a Promise, but a {type(p}).")
```
In many configurations, `assert` statements are skipped